src/main/java/appeng/util/item/AESharedItemStack.java

```diff
@@ -97,7 +98,7 @@ public int compareTo(final AESharedItemStack b) {
             return damageValue;
         }
 
-        return 0;
+        return this.hashCode - b.hashCode;
```

Comment:
This is only intended as a stopgap I presume because hashCode is not _guaranteed_ to be unique, right?

Reply:
Technically no, but that probably applies to anything with a finite number. At least until we reach something like 2^265 and can address every atom in the universe.

It still depends on the item id, which is unique and damage value, which is mostly unique for our use case. The chance of having two NBT objects with the same hashcode is more or less negligible. So in my opinion worth the risk.

Reply:
You are assuming that this is some kind of cryptographic super-hash which it is not, why do we not use identity-hash-code here, which at least is supposedly serial?

Reply:
Oh wait. I assumed it would use the identity hashcode here. I'll change that.

Reply:
`System.identityHashCode(null)` will return `0`, so without a tag it will default to `0 - 0`